test/refactoring/ConvertAsync/basic.swift

```diff
@@ -0,0 +1,189 @@
+func simpleResult(completion: (Result<String, Never>) -> Void) { }
+// ASYNC-SIMPLERESULT: func simpleResult() async -> String { }
+
+// RUN: %refactor -add-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix=ASYNC-ERRORRESULT %s
+func errorResult(completion: (Result<String, Error>) -> Void) { }
+// ASYNC-ERRORRESULT: func errorResult() async throws -> String { }
+
+// RUN: %refactor -add-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix=ASYNC-CUSTOMERRORRESULT %s
+func customErrorResult(completion: (Result<String, CustomError>) -> Void) { }
+// ASYNC-CUSTOMERRORRESULT: func customErrorResult() async throws -> String { }
+
+// RUN: %refactor -add-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix=ASYNC-ALIASRESULT %s
+func aliasedResult(completion: SomeResultCallback) { }
+// ASYNC-ALIASRESULT: func aliasedResult() async throws -> String { }
+
+// RUN: %refactor -add-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix=MANY %s
+func many(_ completion: (String, Int) -> Void) { }
+// MANY: func many() async -> (String, Int) { }
+
+// RUN: %refactor -add-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix=OPTIONAL-SINGLE %s
+func optionalSingle(completion: (String?) -> Void) { }
+// OPTIONAL-SINGLE: func optionalSingle() async -> String? { }
+
+// RUN: %refactor -add-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix=MANY-OPTIONAL %s
+func manyOptional(_ completion: (String?, Int?) -> Void) { }
+// MANY-OPTIONAL: func manyOptional() async -> (String?, Int?) { }
+
+// 2. Check that the next functions do not have async alternatives
+
+// RUN: not %refactor -add-async-alternative -dump-text -source-filename %s -pos=%(line+1):1
+func nonCompletion(a: Int) { }
+
+// RUN: not %refactor -add-async-alternative -dump-text -source-filename %s -pos=%(line+1):1
+func multipleResults(completion: (Result<String, Error>, Result<String, Error>) -> Void) { }
+
+// RUN: not %refactor -add-async-alternative -dump-text -source-filename %s -pos=%(line+1):1
+func completionNotLast(completion: (String) -> Void, a: Int) { }
+
+// RUN: not %refactor -add-async-alternative -dump-text -source-filename %s -pos=%(line+1):1
+func nonVoid(completion: (String) -> Void) -> Int { return 0 }
+
+// RUN: not %refactor -add-async-alternative -dump-text -source-filename %s -pos=%(line+1):1
+func completionNonVoid(completion: (String) -> Int) -> Void { }
+
+// RUN: not %refactor -add-async-alternative -dump-text -source-filename %s -pos=%(line+1):1
+func alreadyThrows(completion: (String) -> Void) throws { }
+
+// 3. Check that the various ways to call a function (and the positions the
+//    refactoring is called from) are handled correctly
```

Comment:
Would be good to add some method examples here too, to make sure the base expr is handled correctly too. Another interesting one to have would be a call to a function with a no-param completion closure that's marked `@autoclosure` as there won't be a closure in the call.